src/codec.rs

```diff
@@ -78,12 +82,20 @@ impl From<Utf8Error> for ParseError {
 /// Encodes and decodes Language Server Protocol messages.
 #[derive(Clone, Debug)]
 pub struct LanguageServerCodec<T> {
+    message_len: Option<usize>,
     _marker: PhantomData<T>,
 }
 
+impl<T> LanguageServerCodec<T> {
+    pub fn reset(&mut self) {
```

Comment:
Is there any reason for this method to be marked `pub`? While `LanguageServerCodec` is not exposed in the crate's public API, I don't think we need to break encapsulation needlessly.

Reply:
Nope, just an oversight. Thanks for catching that.